src/test/java/org/candlepin/pki/certs/ContentAccessPayloadKeyBuilderTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pki.certs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.Environment;
+import org.candlepin.model.Owner;
+import org.candlepin.test.TestUtil;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class ContentAccessPayloadKeyBuilderTest {
```

Comment:
Can we add a couple more tests here? Specifically, tests that check that
- the generated key is the same when the same list of environments but in different order is given
- the generated key is the same when the same list of architectures but in different order in the 'supported_architectures' fact

Reply:
I added more tests around the generated key that also negatively tests that the key is updated with different architectures, supported architectures, and environments.

I didn't add the test that asserts the same key with the same environments in a different order based on the following other comment. It looks like we have to use the environment priority ordering that is returned from the curator.

https://github.com/candlepin/candlepin/pull/5024#discussion_r1956207437

Reply:
Ah, yeah! I will update this and get some tests to assert this logic

Reply:
I added `distinct()` to `ContentAccessPayloadKeyBuilder.getNormalizedEnvironments` and added a unit test to assert these changes